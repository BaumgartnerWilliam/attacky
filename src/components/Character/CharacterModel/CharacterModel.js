import React from 'react';
import { Row } from 'react-bootstrap';
import ApngComponent from 'react-apng'; // replace apngComponent

const CharacterModel = ({ src }) => {
  return (
    <Row data-testid="model" className="character-model">
      <ApngComponent style={{ maxHeight: '200px' }} src={src} autoPlay={true} />
    </Row>
  );
};

export default CharacterModel;
