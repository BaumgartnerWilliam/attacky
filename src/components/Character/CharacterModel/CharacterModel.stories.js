import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import CharacterModel from './CharacterModel';
import idleAsset from '../../../assets/monster_idle.png';

export default {
  title: 'CharacterModel',
  decorators: [withKnobs]
};

export const IdleMonster = () => <CharacterModel src={idleAsset} />;
