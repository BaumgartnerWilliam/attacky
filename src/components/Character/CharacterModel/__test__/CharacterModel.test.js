import React from 'react';
import { render } from '@testing-library/react';
import CharacterModel from '../CharacterModel';
import idleAsset from '../../../../assets/monster_idle.png';

describe('CharacterModel component', () => {
  it('renders without crashing', () => {
    const { container } = render(<CharacterModel />);
    expect(container).toBeInTheDocument();
  });

  it('loads the correct asset', () => {
    const { getByTestId } = render(<CharacterModel src={idleAsset} />);
    const el = getByTestId('model');
    expect(el.firstChild).toHaveAttribute('src', 'monster_idle.png');
  });

  it('matches snapshot', () => {
    const { getByTestId } = render(<CharacterModel src={idleAsset} />);
    const testid = getByTestId('model');
    expect(testid).toMatchSnapshot();
  });
});
