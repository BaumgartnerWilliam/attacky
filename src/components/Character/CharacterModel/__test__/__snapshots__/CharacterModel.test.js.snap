// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterModel component matches snapshot 1`] = `
<div
  class="character-model row"
  data-testid="model"
>
  <canvas
    autoplay=""
    src="monster_idle.png"
    style="max-height: 200px;"
  />
</div>
`;
